gulpfile.js

```diff
@@ -1,87 +1,18 @@
-// Include Gulp and Plugins
-var _           = require('underscore');
-var argv        = require('yargs').argv;
-var colors      = require('colors');
-var fm          = require('front-matter');
-var fs          = require('fs');
-var gulp        = require('gulp');
-var gzip        = require('gulp-gzip');
-var handlebars  = require('handlebars');
-var marked      = require('marked');
-var mkdirp      = require('mkdirp');
-var moment      = require('moment');
-var rename      = require('gulp-rename');
-var requireDir  = require('require-dir');
-var rmrf        = require('rimraf');
-var sass        = require('gulp-sass');
-
-var conf        = require('./conf.json');
-var dir         = requireDir('./tasks/');
-
-
-// Define base folders
-var paths = {
-    pages: './views/pages/',
-    posts: './views/posts/',
-    drafts: './views/drafts/',
-    layouts: './views/',
-    includes: './public/src/js/',
-    sass: './public/src/sass/**/*.scss',
-    dist: './dist/',
-    distCss: './dist/css/'
-}
-
-// Handlebars registerPartial method takes name of partial as first arg
-// and also takes a template source or compiled template as second arg
-// in this case it'll be the source files for each corresponding partial
-handlebars.registerPartial('nav', fs.readFileSync(paths.includes + 'nav.hbs', 'utf8'));
-handlebars.registerPartial('head', fs.readFileSync(paths.includes + 'head.hbs', 'utf8'));
-handlebars.registerPartial('footer', fs.readFileSync(paths.includes + 'footer.hbs', 'utf8'));
-handlebars.registerPartial('recent', fs.readFileSync(paths.includes + 'recent.hbs', 'utf8'));
-
-// Global variable function which takes care of slug
-var hyphenateSlug = function(slug) {
-
-    // Convert slug to lowercase, trim whitespace, replace any /'s with '-'s
-    return slug.toLowerCase().trim().replace(/[^\w\s]/g, '').replace(/\s+/g, '-');
-}
-
-// Specifying the sass file and minified CSS file
-var files = {
-    input: './src/sass/style.scss',
-    output: ' style.min.css'
-}
-
-// Requiring all of the external gulp-tasks
-requireDir('./tasks', {recurse: true });
-
-// // Generate new page
-// gulp.task('new-page', ['new-page'], function(){
-//     console.log('New page');
-// });
-
-// // Generate new post
-// gulp.task('new-post', ['new-post'], function(){
-//     console.log('New post');
-// });
-
-
-// Sass gulp task
-gulp.task('sass-build', function () {
-    return gulp.src(files.input)
-        .pipe(sass({outputStyle: 'compressed'}))
-        .pipe(rename(files.output))
-        .pipe(gulp.dest(paths.distCss));
+// Include Gulp and modular task file
+var gulp = require('gulp');
+var tasks = require('./lib/tasks/index');
+
+// Gulp tasks
+gulp.task('post', function() {
+    console.log('Console Message: Building Post'),
```

Comment:
The console log here is redundant. it would be better placed in the `tasks.post`